src/test/components/GlobalTrack.test.js

```diff
@@ -0,0 +1,131 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// @flow
+
+import * as React from 'react';
+import { Provider } from 'react-redux';
+import { mount } from 'enzyme';
+
+import {
+  changeSelectedThread,
+  hideGlobalTrack,
+} from '../../actions/profile-view';
+import GlobalTrack from '../../components/timeline/GlobalTrack';
+import {
+  getGlobalTracks,
+  getRightClickedTrack,
+} from '../../reducers/profile-view';
+import { getSelectedThreadIndex } from '../../reducers/url-state';
+import { getProfileWithNiceTracks } from '../fixtures/profiles/tracks';
+import { storeWithProfile } from '../fixtures/stores';
+import { getMouseEvent } from '../fixtures/utils';
+
+const LEFT_CLICK = 0;
+const RIGHT_CLICK = 2;
+
+describe('timeline/GlobalTrack', function() {
+  /**
+   *  getProfileWithNiceTracks() looks like: [
+   *    'show [thread GeckoMain process]',
+   *    'show [thread GeckoMain tab]',       <- use this global track.
+   *    '  - show [thread DOM Worker]',
+   *    '  - show [thread Style]',
+   *  ]
+   */
+  function setup() {
+    const trackIndex = 1;
+    const profile = getProfileWithNiceTracks();
+    const store = storeWithProfile(profile);
+    const { getState, dispatch } = store;
+    const trackReference = { type: 'global', trackIndex };
+    const tracks = getGlobalTracks(getState());
+    const track = tracks[trackIndex];
+    if (track.type !== 'process') {
+      throw new Error('Expected a process track.');
+    }
+    const threadIndex = track.mainThreadIndex;
+    if (threadIndex === null) {
+      throw new Error('Expected the track to have a thread index.');
+    }
+
+    // The assertions are simpler if this thread is not already selected.
+    dispatch(changeSelectedThread(threadIndex + 1));
+
+    const view = mount(
+      <Provider store={store}>
+        <GlobalTrack trackIndex={trackIndex} trackReference={trackReference} />
+      </Provider>
+    );
+
+    const globalTrackLabel = view.find('.timelineTrackLabel').first();
+    const globalTrackRow = view.find('.timelineTrackGlobalRow').first();
+
+    return {
+      dispatch,
+      getState,
+      profile,
+      store,
+      view,
+      trackReference,
+      trackIndex,
+      threadIndex,
+      globalTrackLabel,
+      globalTrackRow,
+    };
+  }
+
+  it('matches the snapshot of a global track', () => {
+    const { view } = setup();
+    expect(view).toMatchSnapshot();
+  });
+
+  it('has the correct selectors into useful parts of the component', function() {
+    const { globalTrackLabel, globalTrackRow } = setup();
+    expect(globalTrackLabel.text()).toBe('Content');
+    expect(globalTrackRow.exists()).toBe(true);
+  });
+
+  it('starts out not being selected', function() {
```

Comment:
should it test that there is no "selected" css class on it ?

Reply:
Sounds like a good thing to test, I'll add it.